support-files/mariadb-execstartpre-check.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+#
+# This script purpose is to check if MariaDB
+# configuration has problematic configurations
+# and make more reasonable error messages than
+# it does now
+#
+# To make this script work you need:
+#   * my_print_defaults from MariaDB package
+#   * systemctl from systemd package
+#
+MY_PRINT_DEFAULTS=$(which my_print_defaults)
+
+if [ -n "${MY_PRINT_DEFAULTS}" ]
+then
+    echo "Can't find tool: my_print_defaults"
+fi
+
+#
+# Check with my_print_defaults that any
+# path in configuration is not pointing somewhere
+# we can't write or it is invisible
+#
+# Check with systemctl cat mariadb*.service
+# if user has already altered service file to
+# allow writing to there locations
+#
+my_print_defaults --mysqld | while read -r setting
+do
+    conf_var=${setting%%=*}
+    conf_var=${conf_var#--}
+    conf_val=${setting#*=}
+    conf_problem=""
+
+    # These are the places that systemd
+    # is prevented currently to write or access
+    case "$conf_val" in
+        # ProtectSystem prevents writes to /usr, /boot and /efi
+        /usr*|/boot*|/efi*)
+            conf_problem="ProtectSystem"
+            ;;
+        # ProtectHome prevents access to /home, /root and /run/user
+        /home*|/root*|/run/user*)
+            conf_problem="ProtectHome"
+            ;;
+    esac
+
+    if [ -n "${conf_problem}" ]
+    then
+        echo "Your MariaDB server variable '${conf_var}' points to '${conf_val}' location and currently MariaDB SystemD service file prevents that."
+        echo "To solve configuration problem you should create '/etc/systemd/system/mariadb.service.d/overwrite.conf'"
+        echo -n "In that file you need to configure 'ReadWritePaths'-parameter or turn off "
+
+        if [ "${conf_problem}" = "ProtectSystem" ]
+        then
+            echo "'ProtectSystem'-parameter"
+        else
+            echo "'ProtectHome'-parameter"
+        fi
+
+        echo -e "\nWithout modification of your '${conf_var}' configuration or suggested SystemD service file changes"
+        echo -e "it's most likely your MariaDB database won't start!\n"
+        echo "Read more from systemd.exec(5) man page or https://www.freedesktop.org/software/systemd/man/systemd.exec.html"
+        return 1
```

Comment:
It won't start because you `exit 1`. I still think this is the right way to do it. Should there a way to disable this check (environment variable)?

Reply:
Yes there should because I can believe many finds this annoying.

Reply:
Now there is a way to disable script or systemctl